dspace-api/src/main/java/org/dspace/storage/rdbms/DatabaseUtils.java

```diff
@@ -1374,15 +1381,21 @@ else if (dbms_lc.contains("h2")) // Used for unit testing only
     /**
      * Get a reference to the configured DataSource (which can be used to
      * initialize the database using Flyway).
+     * The DataSource is configured via our ServiceManager (i.e. via Spring).
      * <P>
      * This is NOT public, as we discourage direct connections to the database
      * which bypass Hibernate. Only Flyway should be allowed a direct connection.
      * @return DataSource
      */
     protected static DataSource getDataSource()
     {
-        // DataSource is configured via our ServiceManager (i.e. via Spring).
-        return DSpaceServicesFactory.getInstance().getServiceManager().getServiceByName("dataSource", BasicDataSource.class);
+        Object dataSource = DSpaceServicesFactory.getInstance()
```

Comment:
This may be too lax.  We might request a javax.jdbc.DataSource.  Both the Tomcat and Commons packages implement DataSource, and it seems unlikely that any other pooling packages would not.